src/DataTransferObject/AssignmentDtoCollection.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace App\DataTransferObject;
+
+use ArrayIterator;
+use Countable;
+use IteratorAggregate;
+
+class AssignmentDtoCollection implements Countable, IteratorAggregate
+{
+    /** @var AssignmentDto[] */
+    private $collection = [];
+
+    public function __construct(AssignmentDto ...$assignments)
+    {
+        foreach ($assignments as $assignment) {
+            $this->add($assignment);
+        }
+    }
+
+    public function add(AssignmentDto $dto): self
+    {
+        $this->collection[] = $dto;
+
+        return $this;
+    }
+
+    public function merge(AssignmentDtoCollection $assignments): self
```

Comment:
Suggestion: maybe you could use the AppendIterator:
```
public function merge(AssignmentDtoCollection $assignments):\Iterator
{
    $iterator = new AppendIterator();
    $iterator->append($this);
    $iterator->append($aasignments);
    return $iterator;
}

Reply:
Hm, this is interesting, I never used it before. For me it seems a little bit overkill, plus I'm not sure how would it behave if we want to maintain let's say a unique key in the inner collection (example in `UserDtoCollection` where username is the key). I would leave it as it is for now, but it's worth checking it next time. what do you think?